### ISSUE TITLE
CV03 corrupts SQL when applied to looped sql.

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#4681]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

User @iKwesi found that running `fix` on a statement where the final select target in a loop and contains a trailing comma results in a fix which corrupts the SQL.

This minimal query demonstrates:
```sql
SELECT
    {% for col in ['a', 'b', 'c'] %}
        {{col}},
    {% endfor %}
FROM tbl
```

Results in:
```
> sqlfluff fix test.sql -d bigquery -r CV03 -x f
==== finding fixable violations ====
== [test.sql] FAIL
L:   3 | P:  16 | CV03 | Trailing comma in select statement forbidden
                       | [convention.select_trailing_comma]
==== fixing violations ====
1 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
```

After applying the fix we get:

```sql
SELECT
    {% for col in ['a', 'b', 'c'] %}
        {{col}}{% endfor %}
FROM tbl
```

...which is no longer valid SQL.

### Expected Behaviour

In this case, the rule should still flag the issue - but _it should not attempt the fix_, because it cannot uniquely identify the final comma.

### Observed Behaviour

Fix corrupts SQL.

### How to reproduce

See above.

### Dialect

all - but only really relevant for dialects which allow this (e.g. bigquery).

### Version

Confirmed on 2.0.3

### Configuration

occurs on default config

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
